.buildkite/hooks/pre-exit

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -eu
+CODEPATH=$BUILDKITE_BUILD_CHECKOUT_PATH
+cd $CODEPATH && make clean
```

Comment:
Why introduce this second variable? Why not just cd to `$BUILDKITE_BUILD_CHECKOUT_PATH`?

Variables pointing to path names should always be enclosed in double quotes in shell scripts, especially when the path name comes from somewhere else. Consider what happens in the event that `$BUILDKITE_BUILD_CHECKOUT_PATH` contains spaces or other special characters.

A better way to write this would be `make -C "$BUILDKITE_BUILD_CHECKOUT_PATH" clean`

Reply:
Fixed!

The reason for the second variable was clarity on what the path was, but I removed it